diff --git a/Snappier.Tests/SnappyStreamTests.cs b/Snappier.Tests/SnappyStreamTests.cs
index d66003a..12b3cc0 100644
--- a/Snappier.Tests/SnappyStreamTests.cs
+++ b/Snappier.Tests/SnappyStreamTests.cs
@@ -260,4 +260,19 @@ public class SnappyStreamTests(ITestOutputHelper outputHelper)
 
         Assert.True(decompressed.GetBuffer().AsSpan(0, (int)decompressed.Length).SequenceEqual(originalBytes));
     }
+
+    // https://github.com/brantburnett/Snappier/security/advisories/GHSA-pggp-6c3x-2xmx
+    [Fact(Timeout = 5000)]
+    public void MalformedFrameInput()
+    {
+        byte[] data = [ 0x00, 0x04, 0x00, 0x00, 0x64, 0x4e, 0x6c, 0x71, 0x79, 0x20, 0x77, 0x6f, 0x72, 0x6c, 0x64 ];
+
+        using var src = new MemoryStream(data);
+        using SnappyStream decompressor = new(src, CompressionMode.Decompress);
+
+        using var destination = new MemoryStream();
+
+        // Ensure this throws, no infinite loop
+        Assert.Throws<InvalidDataException>(() => decompressor.CopyTo(destination));
+    }
 }
diff --git a/Snappier/Internal/SnappyStreamDecompressor.cs b/Snappier/Internal/SnappyStreamDecompressor.cs
index e6ebd29..117bf40 100644
--- a/Snappier/Internal/SnappyStreamDecompressor.cs
+++ b/Snappier/Internal/SnappyStreamDecompressor.cs
@@ -98,7 +98,12 @@ internal sealed class SnappyStreamDecompressor : IDisposable
                             }
 
                             int availableChunkBytes = _chunkSize - _chunkBytesProcessed;
-                            if (availableChunkBytes > input.Length)
+                            if (availableChunkBytes <= 0)
+                            {
+                                // Malformed block, decompressor needs more data but there is no more available in the chunk
+                                ThrowHelper.ThrowInvalidDataException("Insufficient data in compressed block.");
+                            }
+                            else if (availableChunkBytes > input.Length)
                             {
                                 _decompressor.Decompress(input);
                                 _chunkBytesProcessed += input.Length;
