diff --git a/internal/command/user_v2_human.go b/internal/command/user_v2_human.go
index 9936f637f..ac412daa1 100644
--- a/internal/command/user_v2_human.go
+++ b/internal/command/user_v2_human.go
@@ -160,50 +160,45 @@ func (c *Commands) AddUserHuman(ctx context.Context, resourceOwner string, human
 	if err != nil {
 		return err
 	}
 
 	if err = c.userValidateDomain(ctx, resourceOwner, human.Username, domainPolicy.UserLoginMustBeDomain); err != nil {
 		return err
 	}
 
-	organizationScopedUsername, err := c.checkOrganizationScopedUsernames(ctx, resourceOwner)
-	if err != nil {
-		return err
-	}
-
 	var createCmd humanCreationCommand
 	if human.Register {
 		createCmd = user.NewHumanRegisteredEvent(
 			ctx,
 			&existingHuman.Aggregate().Aggregate,
 			human.Username,
 			human.FirstName,
 			human.LastName,
 			human.NickName,
 			human.DisplayName,
 			human.PreferredLanguage,
 			human.Gender,
 			human.Email.Address,
-			domainPolicy.UserLoginMustBeDomain || organizationScopedUsername,
+			domainPolicy.UserLoginMustBeDomain,
 			human.UserAgentID,
 		)
 	} else {
 		createCmd = user.NewHumanAddedEvent(
 			ctx,
 			&existingHuman.Aggregate().Aggregate,
 			human.Username,
 			human.FirstName,
 			human.LastName,
 			human.NickName,
 			human.DisplayName,
 			human.PreferredLanguage,
 			human.Gender,
 			human.Email.Address,
-			domainPolicy.UserLoginMustBeDomain || organizationScopedUsername,
+			domainPolicy.UserLoginMustBeDomain,
 		)
 	}
 
 	if human.Phone.Number != "" {
 		createCmd.AddPhoneData(human.Phone.Number)
 	}
 
 	// separated to change when old user logic is not used anymore
@@ -410,20 +405,20 @@ func (c *Commands) changeUserEmail(ctx context.Context, cmds []eventstore.Comman
 		if email.Verified {
 			return append(cmds, user.NewHumanEmailVerifiedEvent(ctx, &wm.Aggregate().Aggregate)), code, nil
 		} else {
 			cryptoCode, err := c.newEmailCode(ctx, c.eventstore.Filter, alg) //nolint:staticcheck
 			if err != nil {
 				return cmds, code, err
 			}
 			if email.URLTemplate == "" {
-				email.URLTemplate = c.loginPaths.DefaultEmailCodeURLTemplate(ctx)
+				email.URLTemplate = c.defaultEmailCodeURLTemplate(ctx)
 			}
 
-			cmds = append(cmds, user.NewHumanEmailCodeAddedEvent(ctx, &wm.Aggregate().Aggregate, cryptoCode.Crypted, cryptoCode.Expiry, email.URLTemplate, email.ReturnCode, ""))
+			cmds = append(cmds, user.NewHumanEmailCodeAddedEventV2(ctx, &wm.Aggregate().Aggregate, cryptoCode.Crypted, cryptoCode.Expiry, email.URLTemplate, email.ReturnCode, ""))
 			if email.ReturnCode {
 				code = &cryptoCode.Plain
 			}
 			return cmds, code, nil
 		}
 	}
 	// only create separate event of verified if email was not changed
 	if email.Verified && wm.IsEmailVerified != email.Verified {
