diff --git a/x/costaking/keeper/hooks_finality.go b/x/costaking/keeper/hooks_finality.go
index 5fab1bfb..759879ec 100644
--- a/x/costaking/keeper/hooks_finality.go
+++ b/x/costaking/keeper/hooks_finality.go
@@ -18,8 +18,12 @@ type HookFinality struct {
 }
 
 // AfterBtcDelegationUnbonded subtracts active satoshis from the costaker reward tracker
-func (h HookFinality) AfterBtcDelegationUnbonded(ctx context.Context, fpAddr sdk.AccAddress, btcDelAddr sdk.AccAddress, fpSecuresBabylon bool, isFpInActiveSet bool, sats uint64) error {
-	if !fpSecuresBabylon || !isFpInActiveSet {
+func (h HookFinality) AfterBtcDelegationUnbonded(ctx context.Context, fpAddr sdk.AccAddress, btcDelAddr sdk.AccAddress, fpSecuresBabylon, isFpActiveInPrevSet, isFpActiveInCurrSet bool, sats uint64) error {
+	if !fpSecuresBabylon || !isFpActiveInPrevSet || !isFpActiveInCurrSet {
+		// It needs to check the fp was active in thre previous set and in it is currently active in the current set for the case where:
+		// 1. the fp was active in the block X
+		// 2. block x+1 btc delegation was unbonded (removes sats)
+		// 3. fp becomes inactive (removes sats twice)
 		return nil
 	}
 
@@ -29,8 +33,8 @@ func (h HookFinality) AfterBtcDelegationUnbonded(ctx context.Context, fpAddr sdk
 }
 
 // AfterBtcDelegationActivated adds new active satoshis to the costaker reward tracker
-func (h HookFinality) AfterBtcDelegationActivated(ctx context.Context, fpAddr sdk.AccAddress, btcDelAddr sdk.AccAddress, fpSecuresBabylon bool, isFpInActiveSet bool, sats uint64) error {
-	if !fpSecuresBabylon || !isFpInActiveSet {
+func (h HookFinality) AfterBtcDelegationActivated(ctx context.Context, fpAddr sdk.AccAddress, btcDelAddr sdk.AccAddress, fpSecuresBabylon, isFpActiveInPrevSet, isFpActiveInCurrSet bool, sats uint64) error {
+	if !fpSecuresBabylon || !isFpActiveInPrevSet {
 		return nil
 	}
 
